src/frontend/src/utils/gantt.utils.ts

```diff
@@ -338,14 +395,44 @@ export const aggregateGanttChanges = (eventChanges: EventChange[], ganttTasks: G
     // Calculate the number of weeks
     const duration = Math.ceil(diffInDays / 7);
 
+    const newProject = updatedEvent.type === 'project';
+
+    const workPackageChanges: RequestEventChange[] = [];
+
+    if (newProject) {
+      updatedEvent.workPackages.forEach((wp) => {
+        workPackageChanges.push({
+          eventId: wp.id,
+          name: wp.name,
+          prevStart: new Date(),
+          prevEnd: new Date(),
+          newStart: wp.start,
+          newEnd: wp.end,
+          duration: wp.end.getTime() - wp.start.getTime(),
+          stage: wp.stage,
+          teamName: task.teamName,
+          workPackageChanges: [],
+          baseWbs: { carNumber: wp.carNumber, projectNumber: wp.projectNumber, workPackageNumber: 0 }
+        });
+      });
+    }
+
     const change: RequestEventChange = {
       eventId: updatedEvent.id,
-      name: task!.name,
-      prevStart: task!.start,
-      prevEnd: task!.end,
+      stage: task.stage,
+      teamName: task.teamName,
+      name: updatedEvent.name,
+      prevStart: task?.start ?? new Date(),
+      prevEnd: task?.end ?? new Date(),
       newStart: updatedEvent.start,
       newEnd: updatedEvent.end,
-      duration
+      duration,
+      createProject: updatedEvent.type === 'project',
+      workPackageChanges,
+      baseWbs:
+        updatedEvent.type === 'project'
+          ? { carNumber: updatedEvent.carNumber, projectNumber: 0, workPackageNumber: 0 }
+          : { carNumber: updatedEvent.carNumber, projectNumber: updatedEvent.projectNumber, workPackageNumber: 0 }
```

Comment:
yeah I guess this is where the type would be added, if we could move the weird logic here, or even change the code to flow better so we could directly put the type of change from the submitted event changes to this, that'd be awesome. But as I type that out it probably feels best to just do it in another ticket, probably good for one of the tech leads so they get experience looking thru this code/ cleaning shit up